monkestation/code/modules/machining/machining_exclusives.dm

```diff
@@ -0,0 +1,229 @@
+// The makeshift pulse gun uses the backpack style. So its better here then in the pulse catagory, since backpack is more cooler about it.
+/obj/item/pulsepack
+	name = "Backpack pulse generator"
+	desc = "A pack that converts extreme amounts of power into pre pulse fusion products. Unable to be minaturized due to a lack of technology."
+	icon = 'icons/obj/weapons/guns/minigun.dmi'
+	worn_icon = 'icons/mob/clothing/back/backpack.dmi'
+	worn_icon_state = "pulsepack"
+	icon_state = "holsteredp"
+	lefthand_file = 'icons/mob/inhands/equipment/backpack_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/backpack_righthand.dmi'
+	inhand_icon_state = "pepperbackpacksec" //close enough
+	slot_flags = ITEM_SLOT_BACK
+	w_class = WEIGHT_CLASS_HUGE
+	var/obj/item/gun/energy/pulse/makeshift/gun
+	var/obj/item/stock_parts/cell/pulsepack/battery
+	var/armed = FALSE //whether the gun is attached, FALSE is attached, TRUE is the gun is wielded.
+	var/overheat = 0
+	var/overheat_max = 1
+	var/heat_diffusion = 0.13
+
+
+/obj/item/pulsepack/Initialize(mapload)
+	. = ..()
+	gun = new(src)
+	battery = new(src)
+	START_PROCESSING(SSobj, src)
+
+/obj/item/pulsepack/Destroy()
+	if(!QDELETED(gun))
+		qdel(gun)
+	gun = null
+	QDEL_NULL(battery)
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/item/pulsepack/process(seconds_per_tick)
+	overheat = max(0, overheat - heat_diffusion * seconds_per_tick)
+
+//ATTACK HAND IGNORING PARENT RETURN VALUE
+/obj/item/pulsepack/attack_hand(mob/living/carbon/user, list/modifiers)
+	if(src.loc == user)
+		if(!armed)
+			if(user.get_item_by_slot(ITEM_SLOT_BACK) == src)
+				armed = TRUE
+				if(!user.put_in_hands(gun))
+					armed = FALSE
+					to_chat(user, span_warning("You need a free hand to hold the gun!"))
+					return
+				update_appearance()
+				user.update_worn_back()
+		else
+			to_chat(user, span_warning("You are already holding the gun!"))
+	else
+		..()
+
+/obj/item/pulsepack/attackby(obj/item/W, mob/user, params)
+	if(W == gun) //Don't need armed check, because if you have the gun assume its armed.
+		user.dropItemToGround(gun, TRUE)
+	else
+		..()
+
+
+/obj/item/pulsepack/dropped(mob/user)
+	. = ..()
+	if(armed)
+		user.dropItemToGround(gun, TRUE)
+
+/obj/item/pulsepack/MouseDrop(atom/over_object)
+	. = ..()
+	if(armed)
+		return
+	if(iscarbon(usr))
+		var/mob/M = usr
+
+		if(!over_object)
+			return
+
+		if(!M.incapacitated())
+
+			if(istype(over_object, /atom/movable/screen/inventory/hand))
+				var/atom/movable/screen/inventory/hand/H = over_object
+				M.putItemFromInventoryInHandIfPossible(src, H.held_index)
+
+
+/obj/item/pulsepack/update_icon_state()
+	icon_state = armed ? "notholsteredp" : "holsteredp"
+	return ..()
+
+/obj/item/pulsepack/proc/attach_gun(mob/user)
+	if(!gun)
+		gun = new(src)
+	gun.forceMove(src)
+	armed = FALSE
+	if(user)
+		to_chat(user, span_notice("You attach the [gun.name] to the [name]."))
+	else
+		src.visible_message(span_warning("The [gun.name] snaps back onto the [name]!"))
```

Comment:
Again `src.` You're gonna have to go through all instances and delete them because I'm not gonna make us both insane with a million comments